pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -58,11 +58,18 @@ type TriggerSpec struct {
 	// +optional
 	DeprecatedGeneration int64 `json:"generation,omitempty"`
 
-	Broker     string          `json:"broker,omitempty"`
+	Broker string `json:"broker,omitempty"`
+
+	// +optional
+	Filter *FilterSelector `json:"filter,omitempty"`
+
 	Subscriber *SubscriberSpec `json:"subscriber,omitempty"`
+}
 
-	Type   string `json:"type,omitempty"`
-	Source string `json:"source,omitempty"`
+type FilterSelector struct {
+	Attributes map[string]string `json:"attributes,omitempty" protobuf:"bytes,1,rep,name=attributes"`
```

Comment:
Why the protobuf annotation?

Reply:
copied from LabelSelector struct... Can take it out if you want